### ISSUE TITLE
Fixing the gaps in the classes IDs for COCO dataset

### ISSUE THREAD

**Issue by eslambakr** [facebookresearch/detr#444]
Hello Everyone,

I have read your clarifications regarding why class number is set to 91 instead of 80 #108.
Motivated by that, I am trying to fix the gaps in the classes IDs for COCO dataset, to make the code more efficient. 
Here you are the steps that I have followed:
1- I created a dictionary in datasets\coco.py, to map the original class IDs to the new ones.
```python
coco_mapping = {
    1:1, 2:2, 3:3, 4:4, 5:5, 6:6, 7:7, 8:8, 9:9, 10:10, 11:11, 13:12, 14:13, 15:14, 16:15, 17:16, 18:17, 19:18,
    20:19, 21:20, 22:21, 23:22, 24:23, 25:24, 27:25, 28:26, 31:27, 32:28, 33:29, 34:30, 35:31, 36:32, 37:33, 38:34
    , 39:35, 40:36, 41:37, 42:38, 43:39, 44:40, 46:41, 47:42, 48:43, 49:44, 50:45, 51:46, 52:47, 53:48, 54:49, 55:50
    , 56:51, 57:52, 58:53, 59:54, 60:55, 61:56, 62:57, 63:58, 64:59, 65:60, 67:61, 70:62, 72:63, 73:64, 74:65, 75:66
    , 76:67, 77:68, 78:69, 79:70, 80:71, 81:72, 82:73, 84:74, 85:75, 86:76, 87:77, 88:78, 89:79, 90:80
}
```
2- Replacing ``` target["labels"] = classes``` in datasets\coco.py by 
```python
     mapped_class = torch.zeros_like(classes)
     for i, cls in enumerate(classes):
                mapped_class[i] = coco_mapping[cls.item()]
     target["labels"] = mapped_class
```
3- Replacing ```num_classes = 20 if args.dataset_file != 'coco' else 91``` in models\detr.py by
```python
    elif args.dataset_file == "coco":
        num_classes = 81
```

But unfortunately, when I compared the two approach's behavior, there is a big drop in the validation accuracy
![Det_COCO_91](https://user-images.githubusercontent.com/25908688/135083318-0cfda059-33b6-4b99-bc5c-ac16ccd46cef.png)
![Det_COCO_80](https://user-images.githubusercontent.com/25908688/135083335-952efe76-0e66-4d55-b6eb-9af0f170f3a3.png)

By inspecting the curves I am suspecting the validation part because the training losses seems similar, however, when I checked the validation codes I think it is correct and no further changes are needed in the evaluation part.

So, Is there any other place you think I have to adapt to fix the classes IDs gap?

Thanks in advance.

**Comment by zhiqwang** [comment]
Just a guess, did you map the id back when you calculated the mAP?
